pallets/manta-sbt/src/tests.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Tests for Manta-SBT
+
+use crate::{
+    mock::{new_test_ext, Balances, MantaSBTPallet, Origin as MockOrigin, Test},
+    Error, ReservedIds,
+};
+use frame_support::{assert_noop, assert_ok, traits::Get};
+use manta_crypto::{
+    merkle_tree::{forest::TreeArrayMerkleForest, full::Full},
+    rand::{CryptoRng, OsRng, RngCore},
+};
+use manta_pay::{
+    config::{
+        utxo::MerkleTreeConfiguration, MultiProvingContext, Parameters, UtxoAccumulatorModel,
+    },
+    parameters::{self, load_transfer_parameters, load_utxo_accumulator_model},
+    test,
+};
+use manta_support::manta_pay::{
+    field_from_id, id_from_field, AssetId, AssetValue, TransferPost as PalletTransferPost,
+};
+
+/// UTXO Accumulator for Building Circuits
+type UtxoAccumulator =
+    TreeArrayMerkleForest<MerkleTreeConfiguration, Full<MerkleTreeConfiguration>, 256>;
+
+lazy_static::lazy_static! {
+    static ref PROVING_CONTEXT: MultiProvingContext = load_proving_context();
+    static ref PARAMETERS: Parameters = load_transfer_parameters();
+    static ref UTXO_ACCUMULATOR_MODEL: UtxoAccumulatorModel = load_utxo_accumulator_model();
+}
+
+pub const ALICE: sp_runtime::AccountId32 = sp_runtime::AccountId32::new([0u8; 32]);
+
+/// Turns vec! into BoundedVec
+macro_rules! bvec {
+	($( $x:tt )*) => {
+		vec![$( $x )*].try_into().unwrap()
+	}
+}
+
+/// Loads the [`MultiProvingContext`].
+#[inline]
+fn load_proving_context() -> MultiProvingContext {
+    parameters::load_proving_context(
+        tempfile::tempdir()
+            .expect("Unable to create temporary directory.")
+            .path(),
+    )
+}
+
+/// Samples a [`Mint`] transaction of `asset` with a random secret.
+#[inline]
+fn sample_to_private<R>(asset_id: AssetId, value: AssetValue, rng: &mut R) -> PalletTransferPost
```

Comment:
Have you ran a test coverage tool before ? I know it's not something we've put into a process but it would be nice to see the coverage overall and of this pallet. This is what we used to do way back https://github.com/Manta-Network/pallet-manta-pay#test-coverage

Reply:
Ive used https://about.codecov.io/ in CI, but tbh it gets annoying to maintain.

I think in this PR running a before and after of the tool you linked seems like a good idea to see if this decreases code coverage fro these changes

Reply:
Yeah we don't have to integrate in the ci right, I think there's an issue already for this https://github.com/Manta-Network/Manta/issues/348
Running it manually will be more than enough.